log/sequencer.go

```diff
@@ -230,12 +237,60 @@ func (s Sequencer) initMerkleTreeFromStorage(ctx context.Context, currentRoot tr
 	return s.buildMerkleTreeFromStorageAtRoot(ctx, currentRoot, tx)
 }
 
-// SequenceBatch wraps up all the operations needed to take a batch of queued leaves
-// and integrate them into the tree.
-// TODO(Martin2112): Can possibly improve by deferring a function that attempts to rollback,
-// which will fail if the tx was committed. Should only do this if we can hide the details of
-// the underlying storage transactions and it doesn't create other problems.
-func (s Sequencer) SequenceBatch(ctx context.Context, logID int64, limit int, guardWindow, maxRootDurationInterval time.Duration) (int, error) {
+// sequencingTask is used by Sequencer to extract (pre-)sequenced LogLeaf
+// entries from the storage and commit to the ordering.
+type sequencingTask interface {
+	// load is invoked once to mine a batch of sequenced entries.
+	load(ctx context.Context, limit int, cutoff time.Time) ([]*trillian.LogLeaf, error)
+	// store is invoked once to send the batch back for potential storage update.
+	store(ctx context.Context, leaves []*trillian.LogLeaf) error
+}
+
+// logSequencingTask is a sequencingTask implementation for Log mode.
+// TODO(pavelkalinnikov): Implement it for Sequenced Log mode similarly.
```

Comment:
"Pre-ordered Log mode"?

Reply:
Yes I think it's important we get the terminology right here as there is a lot of scope for confusion. All logs are sequenced so talking about a sequenced log will create doubt.

Reply:
I settled with the "Pre-ordered Log" term, also defined in #964.